src/e3/anod/qualifiers_manager.py

```diff
@@ -436,6 +564,63 @@ def declare_key_value_qualifier(
                 repr_omit_key=repr_omit_key,
             )
 
+    def declare_key_set_qualifier(
+        self,
+        name: str,
+        description: str,
+        test_only: bool = False,
+        default: str | None = None,
+        choices: list[str] | None = None,
+        repr_alias: str | None = None,
+        repr_in_hash: bool = False,
+        repr_omit_key: bool = False,
+    ) -> None:
+        """Declare a new key set qualifier.
+
+        Declare a key set qualifier to allow it use in the spec. It will have an
+        impact on the build_space and component names.
+
+        A key set qualifier is a 'list' qualifier. They require the user to
+        provide their values as a semi-colon separated list.
+
+        This method cannot be called after the end of the declaration phase.
+
+        :param name: The name of the qualifier. It used to identify it and pass it to
+            the spec.
+        :param description: A description of the qualifier purposes. It is used to
+            make the help/error clearer.
+        :param test_only: By default the qualifier are used by all anod actions
+            (install, build, test...). If test_only is True, then this qualifier is
+            only available for test.
+        :param default: The default value given to the qualifier if no value was
+            provided by the user. If no default value is set, then the user must
+            provide a qualifier value at runtime.
+        :param choices: The list of all authorized values for the qualifier.
+        :param repr_alias: An alias for the qualifier name used by the name generation.
+            By default, the repr_alias is the qualifier name itself.
+        :param repr_in_hash: False by default. If True, the qualifier is included in
+            the hash at the end of the generated name. The result is less readable but
+            shorter.
+        :param repr_omit_key: If True, then the name generation don't display the
+            qualifier name/alias. It only use its value.
+        """
+        if self.is_declaration_phase_finished:
+            raise AnodError(
+                f"{self.origin}: qualifier can only be declared in "
+                " declare_qualifiers_and_components"
+            )
+        if not test_only or self.anod_instance.kind == "test":
+            self.qualifier_decls[name] = KeySetDeclaration(
+                origin=self.origin,
+                name=name,
+                description=description,
+                repr_name=repr_alias,
+                repr_in_hash=repr_in_hash,
+                default=default,
+                choices=choices,
+                repr_omit_key=repr_omit_key,
```

Comment:
I don't understand the condition, here. What I read not test_only means all the case or test, so I don't understand why you do this check?

Reply:
I think it's ok